### PR TITLE
Analytics test

### LINKED ISSUE #233: Analytics: tests
Must cover all edge cases:

- global
- for-publisher and for-advertiser: those require authentication
- by channel
- by channel, for-publisher: requires auth

Can be divided into two:

- a test that checks if all the routes that require auth really do require auth
- a test that checks if the aggregation itself is working properly; to do that, we must have data for multiple channels and multiple publishers beforehand

### PR DESCRIPTION
**Pull request by samparsky**
Fixes #233

### CODE REVIEW

File: test/routes.js [hunk 1]
```diff
@@ -392,3 +392,22 @@ tape('should prevent submitting events for a channel in withdraw period', async
 
 	t.end()
 })
+
+tape('should test analytic auth required routes', async function(t) {
+	const channel = getValidEthChannel()
+	const urls = ['/for-publisher', '/for-advertiser', `for-publisher/${channel.id}`, '/advanced']
+	// Submit a new channel; we submit it to both sentries to avoid 404 when propagating messages
+	await Promise.all([
```
Comment (Ivshti) [inline test/routes.js]: This is not needed, as well as getValidEthChannel
